### ISSUE TITLE
Update logic on Automatic Import deletes Bone Attachment

### ISSUE THREAD

**Issue by rafaela-ip** [godotengine/godot#7942]
**Operating system or device - Godot version:**
Windows 10, Godot 2.1.2, and Blender 2.78c with the Better Collada Exporter.

**Issue description:**

On Blender, I have created a basic model/animation, and exported to Godot.
On Godot, I have Attached a "Bone Attachment" as a child of the Skeleton node, as the documentation says.
Everything works fine, but if I need change anything on my model, when export in Blender, Godot will automatic update everything, and the "Bone Attachment" will be deleted.

I have worked with Unity before, but I have to say that I never had problem updating changes, but I'm having a hard time to understand the best workflow with Godot... how I'm supposed to work with "Bone Attachment", if everytime I update a scene get lost?

Also the documentation is not really useful on this, so if this is not a Bug, I would suggest a better explanation on how deal with that :D

My english is not that good, sorry :D... so here a short video for help -> https://www.youtube.com/watch?v=3fzDFM5AE8g

Thanks <3

**Comment by bojidar-bg** [comment]
I think you should have posted this on QA or forums, since it is a question rather than an issue with the underlying code.

If I remember correctly, once ago we had a system to detect changes to imported scenes and preserve them, but, as you can probably guess, it hardly worked properly, so it was removed.
This means that **no** changes to the imported scene will be persisted.
That's why you should generally use "inherited" scenes, which would get all nodes and data from a base scene (in this case the imported one), and can have additional nodes on top of it.

If you think it is about the documentation, you can reopen it on [godotengine/godot-docs](https://github.com/godotengine/godot-docs), since this repository is only about the C++ code of the engine, editor, and a few selected modules.

**Comment by rafaela-ip** [comment]
If the logic update has removed, this is a big problem and BIG change... OMG i losed so much time because of this -> [Update Logic Documentation](http://docs.godotengine.org/en/stable/tutorials/3d/importing_3d_scenes.html#update-logic) :sob: 

> That's why you should generally use "inherited" scenes, which would get all nodes and data from a base scene (in this case the imported one), and can have additional nodes on top of it.

I'm already doing this, the problem is that "Bone Attachment" ([Documentation](http://docs.godotengine.org/en/stable/classes/class_boneattachment.html?highlight=Bone%20Attachment#description)) needs to be used inside the imported 3D scene... 

Maybe a possible solution would be make "Bone Attachment" works outside of the imported 3D scene, adding a option to the user choose to witch "Skeleton" wants to work, and then you can choose the "Bone"?

**Comment by bojidar-bg** [comment]
@rafaela-ip Sorry for not replying, was doing other stuff so far...

Would you mind opening an issue on https://github.com/godotengine/godot-docs for the Update Logic documentation while at it?

In post-2.1, we usually do this by creating a new "inherited" scene, which inherits the imported scene, and then just add the node we want into it.

Because it is possible to do this, I'm not absolutely sure how useful a boneattachment "skeleton" property would be, but I guess it might have some future, so feel free to discuss.

**Comment by kubecz3k** [comment]
Yeah this is not an issue with bone attachment, more with not proper work-flow. In 3.0 the editor almost forces user to create inherited scene for imported models so don't think any more actions are necessary.

**Comment by Calinou** [comment]
Closing, as the import workflow has changed a lot since Godot 3.0.
